recipes/mpdecimal/2.5.x/test_package/conanfile.py

```diff
@@ -1,19 +1,48 @@
 import os
-from conans import ConanFile, CMake, tools
+import pathlib
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout, CMakeDeps
+from conan.tools.build import can_run
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package"
+    generators = "VirtualRunEnv"
+    test_type = "explicit"
+
+    def layout(self):
+        cmake_layout(self)
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def generate(self):
+
+        tc = CMakeToolchain(self)
+        tc.variables["MPDECIMAL_CXX"] = self.dependencies["mpdecimal"].options.cxx
+        if self.settings.os != "Windows":
+            build_type = self.settings.build_type.value
+            tc.variables[f"CMAKE_RUNTIME_OUTPUT_DIRECTORY_{build_type.upper()}"] = "bin"
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def build(self):
         cmake = CMake(self)
-        cmake.definitions["MPDECIMAL_CXX"] = self.options["mpdecimal"].cxx
         cmake.configure()
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
-            self.run("{} 13 100".format(os.path.join("bin", "test_package")), run_environment=True)
-            if self.options["mpdecimal"].cxx:
-                self.run("{} 13 100".format(os.path.join("bin", "test_package_cpp")), run_environment=True)
+        if can_run(self):
+            if self.settings.os != "Windows":
+                bin_path = pathlib.Path(self.cpp.build.bindirs[0], "bin", "test_package")
```

Comment:
This looks like a bad copy where the package has too many `bin` folders 🤔

Reply:
It's a weird mess, if I remove the bin folder Linux build fails, if I add it the Windows one fails but nothing in the code told me why as the executable was where it was expected.